submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,245 @@
+  zones.
+
+```bash
+    annotations:
+      k8s.ovn.org/node-transit-switch-port-ifaddr: '["169.254.0.3/16"]'
+      k8s.ovn.org/zone-name: global
+    name: cluster1-worker
+
+    annotations:
+      k8s.ovn.org/node-transit-switch-port-ifaddr: '["169.254.0.5/16"]'
+      k8s.ovn.org/zone-name: az2
+    name: cluster2-worker
+```
+
+With the current architecture, Submariner network-plugin-syncer adds routes only in a single zone where the network-plugin-syncer pod runs.
+For example, if network-plugin-syncer is deployed in zone 1 it programs OVN db in zone 1. So only pods in zone 1 nodes will be able to talk
+to other clusters. Pods in zone 2 or zone 3 will not be able to reach remote clusters connected via Submariner.
+
+As part of the proposal, we plan to support both the modes and OVN cluster deployments where interconnect
+is not enabled as well.
+
+### Route APIs
+
+As part of this proposal, we are planning to add two new CRDs in Submariner. The plan is to decouple OVN dataplane programming so that this
+can be moved to OVN, when OVN provides a way to add custom routes. We can propose this CRD or consume the CRD that they provide at a later
+point of time.
+
+#### SubmarinerGwRoute
+
+This CR will be created when a remote endpoint is added and there will be one CR per remote cluster. This CRD has two fields
+
+* NextHops - Specifies the list of next hop to reach the remote cluster, in this case it will be the IP of ovn-k8s-mp0
+  interface, the interface used by OVN for host networking.
+* RemoteCIDR - Specifies the list of  remote CIDRs reachable via the next hop.
+
+This CR will be used by the route agent pod running on the active-Gateway node to program OVN to send the traffic destined to
+remote clusters via the Submariner tunnel.
+
+``` go
+type SubmarinerGwRoute struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    SubmarinerRoutePolicySpec SubmarinerRoutePolicySpec `json:"submarinerRoutePolicySpec"`
+}
+
+type SubmarinerRoutePolicySpec struct {
+    //Specifies the list of next hops to reach the remote CIDRs
+    NextHops []string `json:"nextHops"`
+
+    //Specifies the remote CIDRs available via the next hop
+    RemoteCidr []string `json:"remoteCidr"`
+}
+```
+
+#### SubmarinerNonGWRoute
+
+This CR will be created when a remote endpoint is created and there will be one created per endpoint.
+
+* NextHops - Specifies the list of next hops. In this case ,we will have only one, and it will be the transit switch IP of the zone
+where g/w node is present.
+* RemoteCIDR - Specifies the list of remote CIDRs reachable via this gateway.
+
+* In non-g/w node - If the route-agent pod is not in the same zone as Gateway node zone, send the traffic to the g/w node zone.
+* In g/w node - add route to send the traffic from other zones, destined to remote cluster to the submariner tunnel
```

Comment:
For gw case, can you explain what are the differences between this DP configuration and the one from line 72 above (https://github.com/submariner-io/enhancements/pull/163/files#diff-bd280a9e59e8197a4382d8d40560aa66ac1e49e187beb97aa37b980785990be5R72) ?

Reply:
>Since ovn-cluster router is seperate for each zones, traffic from other zones are by default dropped by ovn cluster router . So this rule is requires to allow traffic from other zones destined to a remote cluster . This is a route .

What information is needed to configure this rule? 
do we have the information needed to configure this tule in NonGatewayRoute CR ?

Reply:
your are right , this need to be done in response to gatewayRoute  CR  changed.